### DISCUSSION TITLE
Empty SignatureHelp leads to wrong-type-argument plistp []

### DISCUSSION THREAD

**Discussion by jdtsmith** [joaotavora/eglot#962] [General]
Using pylsp v1.4.1, when hovering on a function that pylsp doesn't know, signatureHelp is requested, and the response from pylsp is an empty array:

```
(:jsonrpc "2.0" :id 52 :method "textDocument/signatureHelp" :params
	  (:textDocument
	   (:uri "file:///Users/jdsmith/code/python/pahfit/pahfit/features.py")
	   :position
	   (:line 330 :character 42)))
[client-request] (id:53) Sun May 22 21:09:30 2022:
(:jsonrpc "2.0" :id 53 :method "textDocument/hover" :params
	  (:textDocument
	   (:uri "file:///Users/jdsmith/code/python/pahfit/pahfit/features.py")
	   :position
	   (:line 330 :character 42)))
[client-request] (id:54) Sun May 22 21:09:30 2022:
(:jsonrpc "2.0" :id 54 :method "textDocument/documentHighlight" :params
	  (:textDocument
	   (:uri "file:///Users/jdsmith/code/python/pahfit/pahfit/features.py")
	   :position
	   (:line 330 :character 42)))
[stderr] 2022-05-22 21:09:30,260 EDT - WARNING - pylsp.config.config - Failed to load hook pylsp_signature_help: Ran out of input
[server-reply] (id:52) Sun May 22 21:09:30 2022:
(:jsonrpc "2.0" :id 52 :result
	  [])
```

jsonrpc does not like this empty array:

```
Debugger entered--Lisp error: (wrong-type-argument plistp [])
  plist-member([] :signatures)
  #f(compiled-function (jsonrpc-lambda-elem10) #<bytecode -0x71faab247997079>)([])
  jsonrpc-connection-receive(#<eglot-lsp-server eglot-lsp-server-1fe80a1f9620> (:jsonrpc "2.0" :id 8 :result []))
  jsonrpc--process-filter(#<process EGLOT (__)> "Content-Length: 36\15\nContent-Type: application/vsco...")
```

Appreciate any help.

**Comment by joaotavora** [comment]
According to https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#textDocument_signatureHelp, it seems the server is wrong here.  
![image](https://user-images.githubusercontent.com/387011/169731608-72bcc240-e39b-4764-af04-ed1afe20a264.png)
So while there is nothing wrong with returning an empty array from the JSONRPC pov, it seems to be inconsistent with the LSP standard for this particular response.

**Reply by joaotavora** [reply]
To be absolutely certain that it's not the LSP server who is at fault here, more info from the actual JSON message has to be gathered.  

The line

```
 jsonrpc--process-filter(#<process EGLOT (__)> "Content-Length: 36\15\nContent-Type: application/vsco...")
```

of your backtrace is too heavily abbreviated to be able to tell that.  

Please click the `...` or adjust `backtrace-line-length` and try again.

**Comment by jdtsmith** [comment]
Sorry for the delay; see below.  I do suspect this is an underlying server issue, since it occurs only after the server has been running for some time, and always after it throws `Ran out of input` to stderr.  But still, the empty list might show up instead of null sometimes, so guarding against that is probably sensible.  Thanks.

```
Debugger entered--Lisp error: (wrong-type-argument plistp [])
  plist-member([] :signatures)
  #f(compiled-function (jsonrpc-lambda-elem10) #<bytecode -0x1f36912df202c039>)([])
  jsonrpc-connection-receive(#<eglot-lsp-server eglot-lsp-server-1fe9f7e1835a> (:jsonrpc "2.0" :id 1825 :result []))
  jsonrpc--process-filter(#<process EGLOT (pahfit/python-mode)> "Content-Length: 39\15\nContent-Type: application/vscode-jsonrpc; charset=utf8\15\n\15\n{\"jsonrpc\":\"2.0\",\"id\":1825,\"result\":[]}")
```
